pkg/kubelet/kubelet.go

```diff
@@ -1171,7 +1171,9 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
 	var podStatus api.PodStatus
 	if updateType == SyncPodCreate {
 		podStatus = pod.Status
-		glog.V(3).Infof("Not generating pod status for new pod %v", podFullName)
+		podStatus.StartTime = &util.Time{start}
```

Comment:
Yes, setPodStatus does set the time if it's unset. The only reason I am doing it here is that I can reuse the initial timestamp: start. Before initializing PodStatus, kubelet does setup for pod, such as mount. Based on my previous experience, mount could be very slow operation taking many minutes under resource pressure, especially with memory.  

I did consider to set StartTim on the watch notification initially. But like what you mentioned, I have concern with activeDeadline feature.